internal/controller/apiextensions/definition/reconciler.go

```diff
@@ -209,6 +216,14 @@ func WithClientApplicator(ca resource.ClientApplicator) ReconcilerOption {
 	}
 }
 
+// WithClientset specifies how the Reconciler should interact with the
+// Kubernetes API.
```

Comment:
It always bothers me a bit when we have to use both a controller-runtime client and a client-go clientset. I'm guessing its unavoidable in this case right, because the controller-runtime client can't SSA typed objects (only `*Unstructured`)?